tests/cpp/test_resharding.cpp

```diff
@@ -224,6 +224,100 @@ TEST_F(ReshardingTest, Detection) {
   EXPECT_TRUE(isResharding(tv26->definition()));
 }
 
+TEST_F(ReshardingTest, InsertResharding_Before) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* a = makeSymbolicTensor(3);
+  TensorView* b = makeSymbolicTensor(3);
+  TensorView* c = add(a, b);
+  fusion.addInput(a);
+  fusion.addInput(b);
+  fusion.addOutput(c);
+
+  DeviceMesh mesh0({0, 1});
+  DeviceMesh mesh1({2});
+  a->setDeviceMesh(mesh0);
+  b->setDeviceMesh(mesh0);
+  c->setDeviceMesh(mesh1);
+
+  a->axis(0)->parallelize(ParallelType::DIDx);
+  c->axis(1)->parallelize(ParallelType::DIDx);
+
+  insertReshardings(&fusion);
+  std::vector<Val*> outputs = fusion.outputs();
+
+  c = outputs[0]->as<TensorView>();
+  std::vector<TensorView*> inputs(c->definition()->inputs().size());
+  for (auto i : c10::irange(c->definition()->inputs().size())) {
+    inputs[i] = c->definition()->input(i)->as<TensorView>();
+  }
+  EXPECT_TRUE(getTvsWithDifferentSharding(c, inputs).empty());
+}
+
+TEST_F(ReshardingTest, InsertResharding_After) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* a = makeSymbolicTensor(3);
+  TensorView* b = relu(a);
+  fusion.addInput(a);
+  fusion.addOutput(b);
+
+  DeviceMesh mesh0({0, 1});
+  DeviceMesh mesh1({2});
+  a->setDeviceMesh(mesh0);
+  b->setDeviceMesh(mesh1);
+
+  a->axis(0)->parallelize(ParallelType::DIDx);
+  b->axis(1)->parallelize(ParallelType::DIDx);
+
+  insertReshardings(&fusion);
+  std::vector<Val*> outputs = fusion.outputs();
+
+  b = outputs[0]->as<TensorView>();
+  Expr* expr = b->definition();
+  EXPECT_TRUE(expr->isA<LoadStoreOp>());
+  EXPECT_EQ(expr->as<LoadStoreOp>()->opType(), LoadStoreOpType::Set);
+  std::vector<TensorView*> tvs = {expr->inputs()[0]->as<TensorView>()};
+  EXPECT_TRUE(getTvsWithDifferentSharding(a, tvs).empty());
```

Comment:
```suggestion
  EXPECT_THAT(getTvsWithDifferentSharding(a, tvs), IsEmpty());
```

https://google.github.io/googletest/reference/matchers.html

The practical benefit is that the error message will print out the content of the vector if the check fails.